### PR TITLE
[tcgc] update crossLanguageDefinitionId to correspond to apiview

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes https://github.com/Azure/typespec-azure/issues/849

### CODE REVIEW

File: packages/typespec-client-generator-core/test/public-utils.test.ts [hunk 1]
```diff
@@ -921,7 +923,7 @@ describe("typespec-client-generator-core: public-utils", () => {
         const models = runner.context.experimental_sdkPackage.models;
         strictEqual(models.length, 1);
         strictEqual(models[0].name, "TestRequest");
-        strictEqual(models[0].crossLanguageDefinitionId, "TestService.TestRequest");
+        strictEqual(models[0].crossLanguageDefinitionId, "TestService.test.Request.anonymous");
```
Comment (tjprescott) [inline packages/typespec-client-generator-core/test/public-utils.test.ts:926]: I see a lot of `Service.blah.Foo.anonymous` tests in here, what about one for:
```
model Foo {
  props: {
    name: string;
  }
}
```
The cross language id for that name would be `Foo.props.anonymous.name`.

File: packages/typespec-client-generator-core/test/package.test.ts [hunk 2]
```diff
@@ -636,7 +636,7 @@ describe("typespec-client-generator-core: package", () => {
 
       const fooAccessor = mainClient.methods[0];
       strictEqual(fooAccessor.kind, "clientaccessor");
-      strictEqual(fooAccessor.crossLanguageDefintionId, "TestService.getFoo");
+      strictEqual(fooAccessor.crossLanguageDefintionId, "TestService.Foo");
```
Comment (tjprescott) [inline packages/typespec-client-generator-core/test/package.test.ts:639]: Annotating what the X-lang IDs from APIView would be in this test case. Do they jive with your changes?
```
namespace Foo {     // TestService.Foo
  interface Bar {        // TestService.Foo.Bar
    @route("/one")   
    one(): void;          // TestService.Foo.Bar.one
  }
}
interface Bar {        // TestService.Bar
  @route("/two")
  two(): void;          // TestService.Bar.two
}
```

File: packages/typespec-client-generator-core/test/package.test.ts [hunk 3]
```diff
@@ -646,7 +646,7 @@ describe("typespec-client-generator-core: package", () => {
       strictEqual(barAccessor.kind, "clientaccessor");
       strictEqual(barAccessor.access, "internal");
       strictEqual(barAccessor.name, "getBar");
-      strictEqual(barAccessor.crossLanguageDefintionId, "TestService.getBar");
+      strictEqual(barAccessor.crossLanguageDefintionId, "TestService.Bar");
```
Comment (tjprescott) [inline packages/typespec-client-generator-core/test/package.test.ts:649]: Also, should there be an anonymous model test in this file as well?

**Review by tjprescott — COMMENTED** [review]
A few comments.

File: packages/typespec-client-generator-core/test/public-utils.test.ts (hunk 1 above)
Comment (iscai-msft) [reply packages/typespec-client-generator-core/test/public-utils.test.ts:926]: we do have quite a few of those, one example would be line 1541 in the same test file

File: packages/typespec-client-generator-core/test/package.test.ts (hunk 2 above)
Comment (iscai-msft) [reply packages/typespec-client-generator-core/test/package.test.ts:639]: yep they correspond

File: packages/typespec-client-generator-core/test/package.test.ts (hunk 3 above)
Comment (iscai-msft) [reply packages/typespec-client-generator-core/test/package.test.ts:649]: the anonymous tests are all in `public-utils.test.ts` and `types.test.ts`, the `package` test file is more to do with namesapce and interfaces / clients

**Review by tjprescott — APPROVED** [review]

**Pull request merged**